Logstash/logstash-input-azurewadtable/lib/logstash/inputs/azurewadtable.rb

```diff
@@ -23,10 +24,19 @@ class Interrupted < StandardError; end
   # See issue #23 for more: https://github.com/Azure/azure-diagnostics-tools/issues/23
   config :data_latency_minutes, :validate => :number, :default => 1
 
+  # Number of past queries to be run, so we don't miss late arriving data
+  config :past_queries_count, :validate => :number, :default => 5
+
   TICKS_SINCE_EPOCH = Time.utc(0001, 01, 01).to_i * 10000000
 
+  INITIAl_QUERY_SPLIT_PERIOD_MINUTES = 30
+
   def initialize(*args)
     super(*args)
+    if @collection_start_time_utc.nil?
+      @collection_start_time_utc = (Time.now - ( 60 * @data_latency_minutes) - 60).iso8601
```

Comment:
You are going back in time for @data_latency_minutes + 1 minutes. Why the +1 minute?

Reply:
collection_start_time_utc must always be lower than until_timestamp.
until_timestamp is now - data_latency_minutes, so I just chose a time before that. One minute is just a number, it can also be just 1 second, but it has to be less than now - 1 minutes

Reply:
Got it